### PR TITLE
fix: modify some tests

### PR DESCRIPTION
**Pull request by wep21**
modify some header include orders and change some brackets to double quotes to pass the cpplint test.

### CODE REVIEW

**Comment by wep21** [comment]
@jacobperron @j-rivero Could you review this PR?

**Comment by scpeters** [comment]
so the `.hh` headers need to be enclosed in `""` instead of `<>`?

**Comment by jacobperron** [comment]
This overlaps with https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1355

Also, I'm proposing a change to ament_cpplint to fix a bug where .hh files are considered C system headers (which may affect this PR): https://github.com/ament/ament_lint/pull/374

**Comment by jacobperron** [comment]
> so the .hh headers need to be enclosed in "" instead of <>?

I think this is only a workaround, to trick cpplint into thinking they are not C system headers. After https://github.com/ament/ament_lint/pull/374, using double-quotes shouldn't be necessary.

**Comment by wep21** [comment]
How about  once ignoring cpplint include order until https://github.com/ament/ament_lint/pull/374 is merged?

**Comment by jacobperron** [comment]
> How about once ignoring cpplint include order until https://github.com/ament/ament_lint/pull/374 is merged?

Sounds good to me. Thanks for working on it!

File: gazebo_ros/test/test_sim_time.cpp [hunk 1]
```diff
@@ -62,7 +62,7 @@ TEST_F(TestSimTime, TestClock)
   std::vector<ClockMsg::SharedPtr> msgs;
   std::vector<rclcpp::Time> times;
   auto sub = node->create_subscription<ClockMsg>(
-    "/clock", rclcpp::SystemDefaultsQoS(),
+    "/clock", rclcpp::ClockQoS(),
```
Comment (jacobperron) [inline gazebo_ros/test/test_sim_time.cpp:64]: Maybe we should also update the publisher QoS to use the same QoS:

https://github.com/ros-simulation/gazebo_ros_pkgs/blob/9954bb9f5c02e463e1227aa3f756247e8c6756df/gazebo_ros/src/gazebo_ros_init.cpp#L163-L165

File: gazebo_ros/test/test_sim_time.cpp [hunk 2]
```diff
@@ -88,7 +88,7 @@ TEST_F(TestSimTime, TestClock)
       EXPECT_GT(rclcpp::Time(msgs[i]->clock), rclcpp::Time(msgs[i - 1]->clock)) <<
         rclcpp::Time(msgs[i]->clock).nanoseconds() << "    " <<
         rclcpp::Time(msgs[i - 1]->clock).nanoseconds();
-      EXPECT_GT(times[i], times[i - 1]) << times[i].nanoseconds() << "    " <<
+      EXPECT_GE(times[i], times[i - 1]) << times[i].nanoseconds() << "    " <<
```
Comment (wep21) [inline gazebo_ros/test/test_sim_time.cpp]: I don't know how to ensure that test clock callback is executed after rclcpp internal clock callback, so I changed the condition of sim time test.

File: gazebo_ros/test/test_sim_time.cpp (hunk 1 above)
Comment (wep21) [reply gazebo_ros/test/test_sim_time.cpp:64]: @jacobperron I fixed publisher QoS at https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1380/commits/261df1f52c2cf17231bc9dc34fd8050591e1d1ef.

**Comment by wep21** [comment]
@ros-pull-request-builder

**Comment by jacobperron** [comment]
@ros-pull-request-builder retest this please

File: gazebo_ros/test/test_sim_time.cpp (hunk 2 above)
Comment (jacobperron) [reply gazebo_ros/test/test_sim_time.cpp]: I'm not sure the test is well-formed since it depends on the order in which the executor processes callbacks. It might be better to just assert that messages are received on the clock topic instead of also trying to test that the node's use_sim_time logic is working (seems out of scope for this test). So, I'd be okay with just removing this check.

**Review by jacobperron — APPROVED** [review]

**Comment by jacobperron** [comment]
Looks like there are some, unrelated, flaky tests. I'll try triggering CI one more time.

**Comment by jacobperron** [comment]
@ros-pull-request-builder retest this please

File: gazebo_ros/test/test_sim_time.cpp (hunk 2 above)
Comment (wep21) [reply gazebo_ros/test/test_sim_time.cpp]: @jacobperron I removed the test of use_sim_time logic at https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1380/commits/1864f19f02fae83dfbfc777bb8e14cf5570ec4dd.

**Review by jacobperron — APPROVED** [review]

File: gazebo_ros/src/gazebo_ros_init.cpp [hunk 3]
```diff
@@ -162,7 +162,7 @@ void GazeboRosInit::Load(int argc, char ** argv)
   // the simulation is paused), late subscribers can receive the previously published message(s).
   impl_->clock_pub_ = impl_->ros_node_->create_publisher<rosgraph_msgs::msg::Clock>(
     "/clock",
-    rclcpp::QoS(rclcpp::KeepLast(10)).transient_local());
+    rclcpp::ClockQoS());
```
Comment (jacobperron) [inline gazebo_ros/src/gazebo_ros_init.cpp:165]: I'm having second thoughts about this change; I didn't realize the durability setting for `rclcpp::ClockQoS` is volatile (instead of transient local). I think it might be better to revert this change so we keep the same behavior for anyone depending on the behavior from early versions.
